src/jit/lowerxarch.cpp

```diff
@@ -1030,6 +1034,13 @@ void Lowering::TreeNodeInfoInitShiftRotate(GenTree* tree)
     GenTreePtr shiftBy = tree->gtOp.gtOp2;
     GenTreePtr source  = tree->gtOp.gtOp1;
 
+#ifdef _TARGET_X86_
+    if (source->OperGet() == GT_LONG)
+    {
```

Comment:
It would be useful to add a comment here to describe why this is a special case.